### PR TITLE
added forum page and fixed a few front end bugs

### PR DESCRIPTION
**Pull request by NikolasHadjipanayi**
Added forum page.
Fixed a few bugs.

### CODE REVIEW

File: controllers/API/comment.js [hunk 1]
```diff
@@ -38,21 +39,21 @@ router.post("/API/forum/comment", function (req, res) {
 });
 
 router.delete("/API/forum/comment", function (req, res) {
-	Comment.findOne({"_id": req.query.id}, function (err, comment) {
+	Comment.findOne({ "_id": req.query.id }, function (err, comment) {
```
Comment (Daanra) [inline controllers/API/comment.js]: `"object-curly-spacing": ["error", "always"]` this needs to be added to `.eslintrc.js`

File: controllers/API/loginCode.js [hunk 2]
```diff
@@ -19,14 +19,14 @@ router.post("/API/loginCode", auth.isAuthorised("ALTER_LOGIN_CODES"), function (
 router.get("/API/loginCode", function (req, res) {
 	const count = parseInt(req.query.count);
 	delete req.query.count;
-	if(req.query.code) {
+	if (req.query.code) {
```
Comment (Daanra) [inline controllers/API/loginCode.js]: `"space-after-keywords": ["error", "never"]` this needs to be added to `eslintrc.js`

File: public/dist/css/informationpages.css [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+.details {
+	color: #0A9BEB;
+	font-weight: bolder;
+	font-family: serif;
+}
+
+.list-group-item:focus {
+	outline-color: #0A9BEB;
+}
+.pagination-style{
```
Comment (Daanra) [inline public/dist/css/informationpages.css]: I do not know why the CI is not throwing an error, I'll have to fix that later, but the CSS linter throws these errors:
`public/dist/css/informationpages.css`
 14:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 14:17  ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 17:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 17:20  ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 18:9   ✖  Expected single space after ":" with a single-line declaration                    declaration-colon-space-after   
 20:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 20:5   ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 23:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 23:10  ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 26:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 26:10  ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 29:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 29:14  ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 33:1   ✖  Expected selector ".panel:hover" to come before selector "#event-screen .panel"   no-descending-specificity       
 33:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 33:12  ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 37:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 37:14  ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 40:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 40:11  ✖  Expected single space before "{"                                                  block-opening-brace-space-before
 43:1   ✖  Expected empty line before rule                                                   rule-empty-line-before          
 43:13  ✖  Expected single space before "{"                                                  block-opening-brace-space-before

File: public/dist/js/controllers/ForumController.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		};
+		$scope.deleteDisabled = true;
+	};
+	getThreads();
+	resetSelectedThread();
+	$scope.pageSize = 5;
+	$scope.currentPage = 1;
+
+	$scope.deleteComment = function (id) {
+		let ok = confirm("Delete comment?");
```
Comment (Daanra) [inline public/dist/js/controllers/ForumController.js]: Should be a `const`

File: package.json [hunk 5]
```diff
@@ -11,6 +11,7 @@
     "url": "https://github.com/RUGSoftEng/2017-Winter-Summer-School-App"
   },
   "dependencies": {
+    "angular-ui-bootstrap": "^2.5.6",
```
Comment (Daanra) [inline package.json]: Why has this been added? I do not see it referenced anywhere

Comment (NikolasHadjipanayi) [reply package.json]: This was required for the pagination of forumpage.

**Pull request merged**